demosplan/DemosPlanCoreBundle/Entity/Permission/AccessControlPermission.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Entity\Permission;
+
+use DateTime;
+use DemosEurope\DemosplanAddon\Contracts\Entities\AccessControlPermissionInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\CustomerInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\OrgaInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\RoleInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\UuidEntityInterface;
+use demosplan\DemosPlanCoreBundle\Entity\CoreEntity;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * Stores the information about the current phase of a procedure.
+ * Currently there a two phases related to a procedure, therefore this Entity is related to the procedure twice.
+ *
+ * @ORM\Table(name="access_control_permissions")
+ *
+ * @ORM\Entity(repositoryClass="demosplan\DemosPlanCoreBundle\Repository\AccessControlPermissionRepository")
+ */
+class AccessControlPermission extends CoreEntity implements UuidEntityInterface, AccessControlPermissionInterface
+{
+    /**
+     * @ORM\Column(type="string", length=36, options={"fixed":true})
+     *
+     * @ORM\Id
+     *
+     * @ORM\GeneratedValue(strategy="CUSTOM")
+     *
+     * @ORM\CustomIdGenerator(class="\demosplan\DemosPlanCoreBundle\Doctrine\Generator\UuidV4Generator")
+     */
+    protected string $id;
+
+    /**
+     * @ORM\Column(name="permission", type="string", nullable=false)
+     */
+    protected string $permission = '';
+
+    /**
+     * @Gedmo\Timestampable(on="create")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private DateTime $creationDate;
+
+    /**
+     * @var DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private $modificationDate;
+
+    /**
+     * @var DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private $deletionDate;
```

Comment:
regarding to you question via slack: this can be removed, if there is no specialized need for that.

Reply:
Updated in new version, also migration updated demosplan/DemosPlanCoreBundle/DoctrineMigrations/2024/05/Version20240521084035.php